diff --git a/surf.c b/surf.c
index 46aef68..9901b1b 100644
--- a/surf.c
+++ b/surf.c
@@ -872,6 +872,8 @@ loaduri(Client *c, const Arg *arg)
 		rp = realpath(uri, NULL);
 		u = g_strdup_printf("file://%s", rp);
 		free(rp);
+    } else if (*uri == ' ') {
+        u = g_strdup_printf("%s%s", searchengine, uri+1);
 	} else {
 		u = g_strrstr(uri, "://") || g_str_has_prefix(uri, "about:") ? g_strdup(uri)
 		    : g_strdup_printf("http://%s", uri);
